src/database-postgresql.cpp

```diff
@@ -0,0 +1,319 @@
+/*
+Copyright (C) 2014 jjb, James Dornan <james at catch22 dot com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "config.h"
+
+#if USE_POSTGRESQL
+
+#include "database-postgresql.h"
+#include "log.h"
+#include "filesys.h"
+
+
+Database_PostgreSQL::Database_PostgreSQL(Settings &conf)
+{
+	m_conf = conf;
+
+	if (PQisthreadsafe() < 1)
+		infostream << "Postgres backend not thread safe." << std::endl;
```

Comment:
Why these lines?  It doesn't have to be thread safe because the databases are only used from one thread.